lib/src/realm_class.dart

```diff
@@ -972,3 +982,35 @@ class MigrationRealm extends DynamicRealm {
 /// * syncProgress - an object of [SyncProgress] that contains `transferredBytes` and `transferableBytes`.
 /// {@category Realm}
 typedef ProgressCallback = void Function(SyncProgress syncProgress);
+
+/// @nodoc
+class RealmAsyncOpenProgressNotificationsController implements ProgressNotificationsController {
+  final RealmAsyncOpenTaskHandle _handle;
+  RealmAsyncOpenTaskProgressNotificationTokenHandle? _tokenHandle;
+  late final StreamController<SyncProgress> _streamController;
+
+  RealmAsyncOpenProgressNotificationsController._(this._handle);
+
+  Stream<SyncProgress> createStream() {
+    _streamController = StreamController<SyncProgress>.broadcast(onListen: _start, onCancel: _stop);
```

Comment:
no, that is not needed. in fact this could be problematic. These events are raised on a background sync thread and we want the release this thread as soon as possible rather than handling the event  on the same thread. Never use this option for our async callbacks, unless you know you need it.

Reply:
Don't use sync! Especially not on public streams.

Reply:
Why is this a broadcast stream?

Reply:
Yeah, I am using `factory StreamController.broadcast` and there is no other factory to use. I see you point but also why would this be an issue? Do you have a suggestion what else to use here?

Reply:
drop `.broadcast` factory and use ctor directly?

Reply:
done